tests/unit/providers/test_gateway_provider.py

```
@@ -13,12 +24,18 @@
 def _mock_otari_client(platform_mode: bool = False) -> MagicMock:
     mock_client = MagicMock()
     mock_client.platform_mode = platform_mode
-    mock_client.openai = AsyncMock()
+    mock_client.completion = AsyncMock()
     return mock_client
 
 
+def _build_gateway_provider() -> GatewayProvider:
+    with patch("any_llm.providers.otari.otari.AsyncOtariClient", return_value=_mock_otari_client(platform_mode=True)):
+        with pytest.warns(DeprecationWarning, match="gateway.*deprecated"):
```

Ruff (0.15.15)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)